A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/A320_Neo_CDU_MainDisplay.js

```diff
@@ -737,11 +819,25 @@ class A320_Neo_CDU_MainDisplay extends FMCMainDisplay {
                 }
             }
             if (this.currentFlightPhase === FlightPhase.FLIGHT_PHASE_TAKEOFF) {
+                let n1 = this.getThrustTakeOffLimit() / 100;
+                SimVar.SetSimVarValue("AUTOPILOT THROTTLE MAX THRUST", "number", n1);
                 if (this.isAirspeedManaged()) {
                     // getCleanTakeOffSpeed is a final fallback and not truth to reality
                     const speed = isFinite(this.v2Speed) ? this.v2Speed + 10 : this.getCleanTakeOffSpeed();
                     this.setAPManagedSpeed(speed, Aircraft.A320_NEO);
                 }
+
+                //This breaks everything, not sure why (from 1.8.3 update)
+                /* let altitude = Simplane.getAltitudeAboveGround();
+                let n1 = 100;
+                if (altitude < this.thrustReductionAltitude) {
```

Comment:
I think we should check if `this.thrustReductionAltitude` exists first. If not, only then default to something.

Because in mod plane, `thrustReductionAltitude` can be null, if it's not known. See changes in PR [585](https://github.com/flybywiresim/a32nx/pull/585/files).

E.g. `if (thrustReductionAltitude == null || altitude < this.thrustReductionAltitude) {`

Reply:
agreed

Reply:
I think we should change it so it checks the flight phase rather than the altitude.

Reply:
Sounds reasonable. Forgot to mention earlier that should get MSL altitude from Simplane too. But if using flight phase then it doesn't matter anymore.

Reply:
@tyler58546 are you still working on this?